packages/ui/lib/components/Form/AddressInput.tsx

```diff
@@ -0,0 +1,180 @@
+      inputSizeStyle,
+      inputStateStyles,
+      withIcon ? 'pl-10' : undefined
+    )
+
+    const handleResolver = async (address: string) => {
+      if (address.length) {
+        setLoading(true)
+        setError('')
+        setSuccess('')
+        setAddressType('')
+      }
+      const result = await web3Service.resolveName(address)
+      if (result && result.name !== null && result.type === 'address') {
+        setLoading(false)
+        setAddressType(result.type)
+        setSuccess(`It's a valid eth address`)
+        setResolvedName(result.name)
+        return result.address
+      } else if (result && result.address !== null && result.type === 'name') {
+        setLoading(false)
+        setAddressType(result.type)
+        setSuccess(`It's a valid ens name`)
+        setResolvedAddress(result.address)
+        return result.address
+      } else if (result && result.address === null || result && result.name === null  && result.type === 'error'){
+        setLoading(false)
+        setAddressType('error')
+        setError(`It's not a valid ens name or address`)
+      }
+    }
+    return (
+      <>
+        <FieldLayout
+          label={label}
+          size={size}
+          success={success}
+          error={error}
+          description={description}
+        >
+          <div className="flex flex-col">
+            <div className="relative">
+              {withIcon && (
+                <span className="absolute inset-y-0 left-0 flex items-center pl-3 pointer-events-none">
+                  {loadingResolvedAddress ? <Icon size={size} icon={LoadingIcon} /> : <Icon size={size} icon={WalletIcon} />}
+                </span>
+              )}
+              <input
+                {...inputProps}
+                id={label}
+                className={inputClass}
+                {...register(name, {
+                  setValueAs: (value: string) => handleResolver(value),
+                })}
+              />
+            </div>
+          </div>
+        </FieldLayout>
+        <div>
+          {loadingResolvedAddress ? (
+            <span className='text-gray-600'>
+              Loading resolved address or ens...
+            </span>
+          ) : !loadingResolvedAddress && resolvedAddress !== '' && addressType === 'name'  ? (
+            <span className='text-gray-600'>
+              {resolvedAddress}
+            </span>
+          ) : !loadingResolvedAddress && addressType === 'name' && resolvedAddress !== '' && isTruncated ? (
+            <span className='text-gray-600'>
+             {minifyAddress(resolvedAddress)}
+            </span>
+          ) : !loadingResolvedAddress && addressType === 'address' && resolvedName !== '' ? (
+            <span className='text-gray-600'>
+              {resolvedName}
+            </span>
+          ) : addressType === 'error' && error && (
+            <span className='text-rose-700'>Please enter a valid ens or address</span>
+          )}
+        </div>
```

Comment:
```suggestion
        <div>
          {!loadingResolvedAddress &&
          resolvedAddress !== '' &&
          addressType === 'name' ? (
            <span className="text-gray-600">{resolvedAddress}</span>
          ) : !loadingResolvedAddress &&
            addressType === 'name' &&
            resolvedAddress !== '' &&
            isTruncated ? (
            <span className="text-gray-600">
              {minifyAddress(resolvedAddress)}
            </span>
          ) : !loadingResolvedAddress &&
            addressType === 'address' &&
            resolvedName !== '' ? (
            <span className="text-gray-600">{resolvedName}</span>
          ) : (
            addressType === 'error' &&
            error && (
              <span className="text-rose-700">
                Please enter a valid ens or address
              </span>
            )
          )}
        </div>
```

lets remove `Loading resolved address or ens...` because we already have the spinner icon

Reply:
But what if there's an error? There should be a proper error message, right? @kalidiagne